File: packages/amplify_core/lib/src/types/datastore/hub/sync_queries_started_event.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import 'dart:convert';

import 'package:amplify_core/amplify_core.dart';

class SyncQueriesStartedEvent extends DataStoreHubEventPayload {
  final List<String> models;
  SyncQueriesStartedEvent(Map<dynamic, dynamic> serializedData)
      : models = List<String>.from(
          jsonDecode(serializedData['models'] as String) as List,
        );
}

File: packages/amplify_datastore/ios/Internal/AWSDataStorePlugin/Sync/Events/SyncQueriesStartedEvent.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

/// Used as HubPayload for the `SyncQueriesStarted`
public struct SyncQueriesStartedEvent {
    /// A list of all model names for which DataStore has started establishing subscriptions
    public let models: [String]

    public init(models: [String]) {
        self.models = models
    }
}
